nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -602,3 +602,66 @@ To delete a Dynamic Group Membership you would send a `DELETE` request to the de
+    - Evaluation of `instance_1.dynamic_groups` add no benefit to `instance_2.dynamic_groups`: each instance will perform `N+1` queries.
+- `dynamic_groups_cached` - A QuerySet of instance to `DynamicGroup` objects, uses cached member list if available. Ideal for most use cases.
+    - Uses cached member lists for each Dynamic Group that is applicable to the instance's content type. Membership check returns a list (what is available as `dynamic_groups_list_cached`) of applicable Dynamic Groups.
+    - A query (`DynamicGroup.objects.filter(pk__in=dynamic_groups_list_cached)`) is necessary to retrieve a QuerySet of `DynamicGroup` objects.
+    - Ideal for most use cases, performing only `1` query if membership lists are cached.
+    - Evaluation of `instance_1.dynamic_groups_cached` benefits `instance_2.dynamic_groups_cached` as all dynamic group membership lists are cached: `instance_1.dynamic_groups_cached` may perform `N+1` queries, but `instance_2.dynamic_groups_cached` will perform `1` query.
+- `dynamic_groups_list` - List of membership of instance to `DynamicGroup` objects, performs one less database query than `dynamic_groups`.
+    - The internal list used by `dynamic_groups` to retrieve a QuerySet of `DynamicGroup` objects, but saves the final query.
+    - Beneficial if you don't need QuerySet instance of `DynamicGroup` objects, but want to use uncached membership lists on a large amount of objects.
+    - Always performs `N` queries where `N` is the number of Dynamic Groups that are applicable to the instance's content type
+- `dynamic_groups_list_cached` - List of membership of instance to `DynamicGroup` objects, uses cached member list if available. Performs no database queries in optimal conditions.
+    - The internal list used by `dynamic_groups_cached` to retrieve a QuerySet of `DynamicGroup` objects, but saves the final query.
+    - The most optimal way to retrieve a list of `DynamicGroup` objects for an instance: Worst case `instance_1.dynamic_groups_list_cached` will perform `N` queries, but `instance_2.dynamic_groups_list_cached` will perform `0` queries.
+
+### Invalidating/Refreshing the Cache
+
+If you need to invalidate the membership cache for a Dynamic Group, you can do so by running the management command: `nautobot-server refresh_dynamic_group_member_caches`. This will invalidate the cache for all Dynamic Groups.
+
+You can also create a `Job` to run periodically to refresh the cache for particular Dynamic Groups and running on a schedule:
+
+```python
+from nautobot.extras.jobs import Job, ObjectVar
+from nautobot.extras.models import DynamicGroup
+
+class ExampleDynamicGroupMemberCacheRefresh(Job):
+    dynamic_group = ObjectVar(
+        model=DynamicGroup
+    )
+
+    class Meta:
+        name = "Update Dynamic Group Member Cache"
+
+    def run(self, data, commit):
+        DynamicGroup.objects.get(pk=data['dynamic_group']).update_cached_members()
+```
```

Comment:
We might consider publishing this as a system job in keeping with Glenn's suggestion about having a "refresh" button. Perhaps that could be a Job Button? Just spitballing here.

Reply:
We don't have system jobs in 1.6 AFAIK.

Reply:
@bryanculver Riiiight. 🧠